node/node.go

```diff
@@ -0,0 +1,120 @@
+package node
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/log"
+	ethNode "github.com/ethereum/go-ethereum/node"
+	"github.com/status-im/status-go/geth/params"
+	"github.com/status-im/status-go/geth/signal"
+)
+
+// node-related errors
+var (
+	ErrEthServiceRegistrationFailure     = errors.New("failed to register the Ethereum service")
+	ErrWhisperServiceRegistrationFailure = errors.New("failed to register the Whisper service")
+	ErrLightEthRegistrationFailure       = errors.New("failed to register the LES service")
+	ErrNodeMakeFailure                   = errors.New("error creating p2p node")
+	ErrNodeRunFailure                    = errors.New("error running p2p node")
+	ErrNodeStartFailure                  = errors.New("error starting p2p node")
+)
+
+type Node interface {
+	Config() *params.NodeConfig
+	Node() *ethNode.Node
+	Start() (chan struct{}, error)
+}
+
+type statusNode struct {
+	sync.Mutex
+	config  *params.NodeConfig
+	node    *ethNode.Node
+	started chan struct{}
+	stopped chan struct{}
+}
+
+func New(config *params.NodeConfig) (Node, error) {
+	stackConfig := defaultEmbeddedNodeConfig(config)
+	n, err := ethNode.New(stackConfig)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &statusNode{
+		config: config,
+		node:   n,
+	}, nil
+}
+
+func (sn *statusNode) Start() (chan struct{}, error) {
```

Comment:
Let's look at how this `statusNode` can be used.

```go
node := node.New(config)
started, err := node.Start()
if err != nil {
  // handle err
}

select {
case <-started:
  // node started but... how can I be sure it's still running? 
  // In `start()` it can fail but only the signal is sent, 
  // `stopped` channel is still open. Even if `stopped` channel was closed 
  // how can I figure out why?
}
```

I would recommend:
1. Removing `signal` package calls from `node`. It's something that the consumer of `node` package should implement,
2. Making `Start` do `sn.node.Start()` synchronously,
3. Implement `Wait() error` which would wait until the node is stopped. It can return an error if the node is not started or already errored between `Start` and `Wait` calls,
4. Implement `Stop() error` that can stop the node. `Wait() error` would return `nil` if the node was stopped successfully through `Stop()` call.

Reply:
thank you @adambabik I'll follow the other PR and keep in mind these suggestions!